### PR TITLE
PreImageOracle: Change local context to bytes32

### PR DESCRIPTION
**Pull request by ajsutton**
**Description**

The local context value used for local inputs in the pre-image oracle are actually being generated as a hash rather than a number. To avoid everything jumping through the hoops of interpreting a hash as a number, change the type from `uint256` to `bytes32`.

**Tests**

Updated tests to use the new type but shouldn't see any change in behaviour (beyond type checkers).

**Metadata**

- https://github.com/ethereum-optimism/client-pod/issues/43

### CODE REVIEW

**Comment by refcell** [comment]
Looks like this needs a `.gas-snapshot` update @ajsutton. (Though note this will probably disappear, at least in its current form, in the future as discussed in slack)

File: packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol [hunk 1]
```diff
@@ -82,8 +82,8 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
     bool internal subgameAtRootResolved;
 
     /// @notice Semantic version.
-    /// @custom:semver 0.0.11
-    string public constant version = "0.0.11";
+    /// @custom:semver 0.0.12
+    string public constant version = "0.0.12";
```
Comment (refcell) [inline packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:86]: Very nice catch.

**Review by refcell — APPROVED** [review]
This looks good to me, will always be able to coalesce solidity uints into bytes32. Though, would like to get @clabby to verify since he has more context on the impact of these changes for FPA+beyond contracts.

File: packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol (hunk 1 above)
Comment (ajsutton) [reply packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:86]: These are the kinds of things a real seasoned contract developer gets right... once the tests fail and they go searching through a bunch of docs to work out what's going on, then have about 5 gos at it. 😆

**Review by clabby — APPROVED** [review]
LGTM

**Pull request merged**